src/main/java/org/apache/mesos/scheduler/SchedulerDriverFactory.java

```diff
@@ -0,0 +1,126 @@
+package org.apache.mesos.scheduler;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.mesos.MesosSchedulerDriver;
+import org.apache.mesos.Scheduler;
+import org.apache.mesos.SchedulerDriver;
+import org.apache.mesos.Protos.Credential;
+import org.apache.mesos.Protos.FrameworkInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.google.protobuf.ByteString;
+
+/**
+ * Factory class for creating {@link MesosSchedulerDriver}s.
+ */
+public class SchedulerDriverFactory {
+
+  private static final Logger LOGGER =
+    LoggerFactory.getLogger(SchedulerDriverFactory.class);
+
+  /**
+   * If this environment variable is present in the scheduler environment, the master is using
+   * some form of sidechannel auth. When this environment variable is present, we should always
+   * provide a {@link Credential} with (only) the principal set.
+   */
+  private static String SIDECHANNEL_AUTH_MAGIC_ENVNAME = "DCOS_SERVICE_ACCOUNT_CREDENTIAL";
+
+  /**
+   * Creates and returns a new {@link SchedulerDriver} without a credential secret.
+   *
+   * @param scheduler The Framework {@link Scheduler} implementation which should receive callbacks
+   *     from the {@link SchedulerDriver}
+   * @param frameworkInfo The {@link FrameworkInfo} which describes the framework implementation.
+   *     The 'principal' field MUST be populated and non-empty
+   * @param masterZkUrl The URL of the currently active Mesos Master, of the form "zk://host/mesos"
```

Comment:
Could we please rename masterZkUrl to something less implementation specific.  Zookeeper is potentially a transient implementation detail.  Maybe `masterUrl`.